pkg/state/state_nv_ipam_cni_test.go

```diff
@@ -21,33 +21,347 @@ import (
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	admv1 "k8s.io/api/admissionregistration/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/log"
 
+	"github.com/Mellanox/network-operator/pkg/config"
 	"github.com/Mellanox/network-operator/pkg/state"
+	staticconfig_mocks "github.com/Mellanox/network-operator/pkg/staticconfig/mocks"
 
 	mellanoxv1alpha1 "github.com/Mellanox/network-operator/api/v1alpha1"
 	"github.com/Mellanox/network-operator/pkg/staticconfig"
 )
 
 var _ = Describe("NVIPAM Controller", func() {
-	ctx := context.Background()
+	var (
+		nvIpamState state.State
+		catalog     state.InfoCatalog
+		client      client.Client
+		namespace   string
+		renderer    state.ManifestRenderer
+	)
 
-	imageSpec := getTestImageSpec()
-	imageSpec = addContainerResources(imageSpec, "nv-ipam-node", "5", "3")
-	imageSpec = addContainerResources(imageSpec, "nv-ipam-controller", "5", "3")
+	BeforeEach(func() {
+		scheme := runtime.NewScheme()
+		Expect(mellanoxv1alpha1.AddToScheme(scheme)).NotTo(HaveOccurred())
+		Expect(v1.AddToScheme(scheme)).NotTo(HaveOccurred())
+		Expect(appsv1.AddToScheme(scheme)).NotTo(HaveOccurred())
+		Expect(admv1.AddToScheme(scheme)).NotTo(HaveOccurred())
+		Expect(rbacv1.AddToScheme(scheme)).NotTo(HaveOccurred())
+		client = fake.NewClientBuilder().WithScheme(scheme).Build()
+		manifestDir := "../../manifests/state-nv-ipam-cni"
+		s, r, err := state.NewStateNVIPAMCNI(client, manifestDir)
+		Expect(err).NotTo(HaveOccurred())
+		nvIpamState = s
+		renderer = r
+		catalog = getTestCatalog()
+		catalog.Add(state.InfoTypeStaticConfig,
+			staticconfig.NewProvider(staticconfig.StaticConfig{CniBinDirectory: "custom-cni-bin-directory"}))
+		namespace = config.FromEnv().State.NetworkOperatorResourceNamespace
+	})
+
+	Context("Verify objects rendering", func() {
+		It("should create Daemonset - minimal spec", func() {
```

Comment:
I know we don't have this in our other tests, but should we start adding negative tests?

Reply:
NB: this is just a reminder, nothing related to this PR :)